api/src/main/java/org/openmrs/module/dss/api/action/SymptomFormAction.java

```diff
@@ -0,0 +1,71 @@
+package org.openmrs.module.dss.api.action;
+
+import org.openmrs.Concept;
+import org.openmrs.Encounter;
+import org.openmrs.Obs;
+import org.openmrs.api.EncounterService;
+
+import org.openmrs.api.context.Context;
+import org.openmrs.module.htmlformentry.CustomFormSubmissionAction;
+import org.openmrs.module.htmlformentry.FormEntrySession;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+public class SymptomFormAction implements CustomFormSubmissionAction {
+
+    private EncounterService encounterService;
+
+    private static final String TEST_CLASS = "Test";
+    private static final Integer DESTINATION_CONCEPT_ID = 166111;
+    private static final String AIMODEL_DESTINATION = "AI END POINT URL";
+
+    private Log log = LogFactory.getLog(this.getClass());
+    
+    public Log getLog() {
+        return log;
+    }
+
+   
+    @Override
+    public void applyAction(FormEntrySession formSession) {
+        Encounter symptomEncounter = formSession.getEncounter();
+    
+        encounterService = Context.getEncounterService();
+    
+        if (symptomEncounter.getObs() != null) {
+            Optional<Obs> destinationObs = symptomEncounter.getObs().stream()
+                    .filter(obs -> obs.getConcept().getConceptId().equals(DESTINATION_CONCEPT_ID)).findFirst();
+            if (destinationObs.isPresent()) {
+                if (destinationObs.get().getValueText().equals(AIMODEL_DESTINATION)) {
+                    Set<Concept> testConcepts = new HashSet<>();
+                    List<Obs> capturedObsList = new ArrayList<>();
+    
+                    symptomEncounter.getObs().forEach(obs -> {
+                        if (obs.getValueCoded() != null) {
```

Comment:
What are trying to do here ??

Reply:
Thanks for the review, Basing on the last chat, you had advised me to use AOP for after returning advice.  Here is the current Pr

Reply:
i see. 
the you should remove the action class

Reply:
Yes i updated the code.